core/drivers/gic.c

```diff
@@ -356,12 +354,18 @@ static void gic_op_enable(struct itr_chip *chip, size_t it)
 {
 	struct gic_data *gd = container_of(chip, struct gic_data, chip);
 
+	if (it >= gd->max_it)
+		panic_trace("invalid it: %d >= %d", it, gd->max_it);
+
 	gic_it_enable(gd, it);
 }
 
 static void gic_op_disable(struct itr_chip *chip, size_t it)
 {
 	struct gic_data *gd = container_of(chip, struct gic_data, chip);
 
+	if (it >= gd->max_it)
+		panic_trace("invalid it: %d >= %d", it, gd->max_it);
```

Comment:
actually, i don't like these panic traces.
driver could easily report a error/ok return value and complain on mis-configurations.